tools/_functions.sh

```diff
@@ -370,6 +370,21 @@ function ensure_webpack_bundle_exists {
     echo "✔ Webpack bundle is in place" | print_success
 }
 
+function ensure_node_modules_exist {
+    if [ ! -d "${BASE_DIR}/node_modules" ]; then
+        echo "Missing node_modules - running npm install..." | print_info
+
+        if ! npm install > /dev/null 2>&1; then
+            echo "Error during npm install" | print_error
+            exit 1
+        fi
+
+        echo "✔ npm install completed successfully" | print_success
```

Comment:
I think we should rather do

```suggestion
        echo "Missing node_modules - please run ./tools/install.sh" | print_info
        exit 1
```

(or something similar). The install script does a bunch of additional checks, incl. correct versions of node and npm.

Maybe we could however add a `--node` flag to that script, similar to the existing `--python` flag, which would exit the script after installing node dependencies?

Reply:
Sounds good @charludo! ☺️